base/gmp.jl

```diff
@@ -346,23 +346,27 @@ for (fJ, fC) in ((:-, :neg), (:~, :com))
     end
 end
 
-function <<(x::BigInt, c::Int)
-    c < 0 && throw(DomainError())
+function <<(x::BigInt, c::UInt)
```

Comment:
Ah, I misread it (I thought it was `Cuint`).

What if `Culong != UInt` (as I think it is on Windows 64-bit): won't the implicit conversion throw an error for `typemax(UInt)`?